File: src/main/java/dk/snemarch/xmltest/ApplicationResponseExtractor.java
package dk.snemarch.xmltest;

import javax.xml.stream.XMLStreamException;
import java.io.InputStream;
import java.util.UUID;

public class ApplicationResponseExtractor extends DataExtractorBase {
	public ExtraDataApplicationResponse extract(InputStream document) throws XMLStreamException {
		ApplicationResponseContentHandler handler = new ApplicationResponseContentHandler();
		super.extractAndSave(document, handler);

		ExtraDataApplicationResponse data = handler.getData();
		data.setDocumentId(UUID.randomUUID());
		return data;
	}
}

File: src/main/kotlin/dk/snemarch/xmltest/ApplicationResponseExtractor.kt
package dk.snemarch.xmltest

import java.io.InputStream
import javax.xml.stream.XMLInputFactory
import javax.xml.stream.XMLStreamException

class ApplicationResponseExtractor(factory: XMLInputFactory) : DataExtractorBase(factory), DataExtractor<ExtraDataApplicationResponse> {
	@Throws(XMLStreamException::class)
	override fun extract(document: InputStream): ExtraDataApplicationResponse {
		val handler = ApplicationResponseContentHandler()
		super.extractAndSave(document, handler)
		return handler.data!!
	}
}
